### DISCUSSION TITLE
Getting continuous Ekka(AutoCluster) warning logs in emqx

### DISCUSSION THREAD

**Discussion by Aravindms369** [emqx/emqx#8419] [Q&A]
Hi @zmstone

emqx version:4.4.4

I am continuously getting the below warning logs in emqx:

[warning] Ekka(AutoCluster): discovery did not succeed; retrying in 5000 ms

Please help to resolve the same.

**Comment by zmstone** [comment]
Hi please share your emqx.conf cluster configs.
Which auto cluster strategy is in use?

**Reply by Aravindms369** [reply]
We have below strategy for cluster config file on emqx:

       - name: EMQX_CLUSTER__DISCOVERY
          value: k8s
        - name: EMQX_CLUSTER__K8S__APISERVER
          value: ****
        - name: EMQX_CLUSTER__K8S__SUFFIX
          value: ****
        - name: EMQX_CLUSTER__K8S__NAMESPACE
          value: ****
        - name: EMQX_CLUSTER__K8S__SERVICE_NAME
          value: ****
        - name: EMQX_CLUSTER__K8S__ADDRESS_TYPE
          value: ****
        - name: EMQX_CLUSTER__K8S__APP_NAME
          value: ****

**Reply by zmstone** [reply]
These values are no secret, could you provide them?

Do you see more logs than below?

> [warning] Ekka(AutoCluster): discovery did not succeed; retrying in 5000 ms

What’s the node name?
is it the same as `EMQX_CLUSTER__K8S__APP_NAME`?

**Reply by Aravindms369** [reply]
Do you see more logs than below? --> The same logs are repeated continuosly, no other logs are related to this

EMQX_CLUSTER__K8S__APP_NAME and node name is different

**Reply by zmstone** [reply]
that's probably the issue.
please make sure `EMQX_CLUSTER__K8S__APP_NAME` is the same as EMQX node name. e.g. if it's `emqx@k8s.host.name`, then the k8s app name has to be `emqx`.

in 5.0, this config is no longer required.

**Comment by Aravindms369** [comment]
@zmstone I have verified again, seems like the app name and node name is same. Both are emqx only

**Reply by Aravindms369** [reply]
Below logs are also getting along with the warning logs:

[info] Ekka(AutoCluster): all discovered nodes already in cluster; ignoring
[debug] Ekka(AutoCluster): join result: ignore
[info] Ekka(AutoCluster): no discovered nodes outside cluster
[warning] Ekka(AutoCluster): discovery did not succeed; retrying in 5000 ms

**Reply by Aravindms369** [reply]
[error] supervisor: 'esockd_connection_sup - <0.1850.0>', errorContext: connection_shutdown, reason: {ssl_error,{tls_alert,{certificate_required,"TLS server: In state wait_cert at tls_handshake_1_3.erl:1475 generated SERVER ALERT: Fatal - Certificate required\n certificate_required"}}}, offender: [{pid,<0.11365.70>},{name,connection},{mfargs,{emqx_connection,start_link,[[{deflate_options,[]},{max_conn_rate,500},{active_n,100},{zone,external},{proxy_address_header,<<>>},{proxy_port_header,<<>>},{supported_subprotocols,[]}]]}}]

[error] supervisor: 'esockd_connection_sup - <0.1850.0>', errorContext: connection_shutdown, reason: {ssl_error,{tls_alert,{handshake_failure,"TLS server: In state certify at tls_dtls_connection.erl:316 generated SERVER ALERT: Fatal - Handshake Failure\n no_client_certificate_provided"}}}, offender: [{pid,<0.3667.75>},{name,connection},{mfargs,{emqx_connection,start_link,[[{deflate_options,[]},{max_conn_rate,500},{active_n,100},{zone,external},{proxy_address_header,<<>>},{proxy_port_header,<<>>},{supported_subprotocols,[]}]]}}]

[error] supervisor: 'esockd_connection_sup - <0.1850.0>', errorContext: connection_shutdown, reason: {ssl_error,{tls_alert,{unexpected_message,"TLS server: In state hello at tls_record.erl:558 generated SERVER ALERT: Fatal - Unexpected Message\n {unsupported_record_type,18}"}}}, offender: [{pid,<0.27317.76>},{name,connection},{mfargs,{emqx_connection,start_link,[[{deflate_options,[]},{max_conn_rate,500},{active_n,100},{zone,external},{proxy_address_header,<<>>},{proxy_port_header,<<>>},{supported_subprotocols,[]}]]}}]

**[error] Ekka(AutoCluster): Discovery error: {500,"{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":****{},\"status\":\"Failure\",\"message\":\"etcdserver: leader changed\",\"code\":500}\n"}**

@zmstone These error logs are also observed along with the logs.

Please help to resolve the same.

**Reply by zmstone** [reply]
> [error] Ekka(AutoCluster): Discovery error: {500,"{"kind":"Status","apiVersion":"v1","metadata":****{},"status":"Failure","message":"etcdserver: leader changed","code":500}\n"}

This seems to be a 500 error returned from k8s API.
I'm afraid my skills are very limited in this area.

**Reply by Aravindms369** [reply]
#8418 

@zmstone This issue is same as the issue I have mentioned here.

[info] Ekka(AutoCluster): all discovered nodes already in cluster; ignoring
[debug] Ekka(AutoCluster): join result: ignore
[info] Ekka(AutoCluster): no discovered nodes outside cluster
[warning] Ekka(AutoCluster): discovery did not succeed; retrying in 5000 ms

**Reply by zmstone** [reply]
Hi @Aravindms369 
That issue is a bit different.
From what's reported, they were able to make two nodes join the cluster when the k8s worker nodes are from the same cloud service provider. But if the k8s workers are from different cloud providers, the nodes could not join. My suspicion was network issue.

> [error] Ekka(AutoCluster): Discovery error: {500,"{"kind":"Status","apiVersion":"v1","metadata":****{},"status":"Failure","message":"etcdserver: leader changed","code":500}\n"}

In your case, since there is a `500` error observed, there should be some logs from k8s ?

**Reply by zmstone** [reply]
or it actually hinted etcd leader-change, maybe that hints some issue on k8s's etcd service ?

**Reply by zmstone** [reply]
Hi again.

Maybe you can switch to `debug` level logging,
it will print more logs to indicate failure reasons etc.

**Reply by zmstone** [reply]
> Hi again.
> 
> Maybe you can switch to `debug` level logging, it will print more logs to indicate failure reasons etc.

Sorry, I did not notice you already have debug level logging.

**Reply by Aravindms369** [reply]
HI @zmstone  
Did you get any idea? debug log level is also enabled here.
